Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py

```diff
@@ -6,78 +6,77 @@
-    sha1[:] = (value for value in sha1 if value != ex_indicator)
-    sha256[:] = (value for value in sha256 if value != ex_indicator)
-    cve[:] = (value for value in cve if value != ex_indicator)
-
-# Convert domain indicators to url (match Domain Indicator Type formatting script)
-domain = ["hxxp://" + sub for sub in domain]
-
-# TLD exclusion for domain indicators
-for indicator in domain:
-    for ex_tld in TLDExclusion:
-        if indicator.endswith(ex_tld):
-            badDomainTLD.append(indicator)
-
-# TLD validation for domain regex
-if unescapeDomain == "True":
-    for dn in domain:
-        tldCheck = get_tld(dn, fail_silently=True)
-        if not tldCheck:
-            badDomainTLD.append(dn)
-
-# Combine all indicators
-blogIndicators = [*md5, *sha1, *sha256, *domain, *url, *ip, *cve]
-
-# Remove bad formatted indicators
-for fp_indicator in badDomainTLD:
-    blogIndicators[:] = (value for value in blogIndicators if value != fp_indicator)
-
-# Keep unique values
-blogIndicators = list(dict.fromkeys(blogIndicators))
-blogIndicators = str(blogIndicators)
-
-return_results(CommandResults(readable_output=blogIndicators, outputs={
-               "http.parsedBlog.indicators": blogIndicators, "http.parsedBlog.sourceLink": blog_url}))
+
+def strip_HTML_tags(page):
+    # Parse the HTML content
+    soup = BeautifulSoup(page.content, "html.parser")
+    # Strip irrelevant tags
+    for data in soup(['style', 'script', 'header', 'head', 'footer', 'aside', 'a']):
+        data.decompose()
+    return(' '.join(soup.stripped_strings))
+
+
+def validate_domains(domains, unescapeDomain, TLDExclusion):
+    # TLD exclusion and validation for domain indicators
+    badDomainTLD = set()
+    for indicator in domains:
+        if unescapeDomain == "True" and not (get_tld(indicator, fail_silently=True)):
+            badDomainTLD.add(indicator)
+            continue
+
+        for tld in TLDExclusion:
+            if indicator.endswith(tld):
+                badDomainTLD.add(indicator)
+    return(badDomainTLD)
+
+
+def main():
+    # Retrieve demisto args
+    args = demisto.args()
+    blog_url = args.get("url")
+    try:
+        headers = {'user-agent': 'PANW-XSOAR'}
+        page = requests.get(blog_url, verify=False, headers=headers)
+        page.raise_for_status
+    except requests.HTTPError:
+        raise
+
+    exclusionList = set(argToList(args.get("exclude_indicators")))
+    TLDExclusion = argToList(args.get("exclude_TLD"))
+    unescapeDomain = args.get("unescape_domain")
+
+    # Allow domain regex replacement between "[.]" and "."
+    if unescapeDomain == "False":
+        domain_regex = r"([a-zA-Z0-9]+?\[?\.?\]?[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]\[\.\][a-zA-Z]{2,}\[?\.?\]?[a-zA-Z]{0,})"
+    else:
+        domain_regex = r"([a-zA-Z0-9]+?\[?\.?\]?[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]\[\.\][a-zA-Z]{2,}\[?\.?\]?[a-zA-Z]{0,})"
+        domain_regex = domain_regex.replace("\[\.\]", "\.")
```

Comment:
Why aren't you using the indicator regex from `CommonServerPython`?

Reply:
I did where I could. 
As far as I know, there is no domain regex in commonServerPython, and the other regexes weren’t good enough to work with.